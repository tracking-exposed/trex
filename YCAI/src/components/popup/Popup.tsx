import {
  Box,
  Button,
  Card,
  CardActions,
  CardContent,
  makeStyles,
  Typography,
} from '@material-ui/core';
import { Alert, AlertTitle } from '@material-ui/lab';
import * as QR from 'avenger/lib/QueryResult';
import { WithQueries } from 'avenger/lib/react';
import { formatDistance } from 'date-fns';
import parseISO from 'date-fns/parseISO';
import React from 'react';
import { useTranslation } from 'react-i18next';
import { config } from '../../config';
import { settings } from '../../state/popup.queries';
import { PopupErrorBox } from './PopupErrorBox';
import Settings from './Settings';

const useStyles = makeStyles((theme) => ({
  container: {
    width: 500,
    padding: theme.spacing(2),
    boxSizing: 'border-box',
  },
  content: {
    marginBottom: theme.spacing(2),
  },
  header: {
    marginTop: theme.spacing(2),
    marginBottom: theme.spacing(0),
  },
  version:{
    marginBottom: theme.spacing(7),
  },
  img: {
    width: '280%',
    maxWidth: 280,
    display: 'block',
  },
  link: {
    color: 'black',
    textDecoration: 'none',
    display: 'block',
  },
  switchFormControl: {
    margin: 0,
  },
  dashboardButton: {
    backgroundColor: theme.palette.common.black,
    padding: theme.spacing(3),
    '& span': {
      lineHeight: 1,
    },
  },
}));

const PopupLoader: React.FC = () => {
  return (
    <Alert severity="info">
      <AlertTitle>Loading settings...</AlertTitle>
      <strong>
        Access{' '}
        <a href="https://www.youtube.com" target="_blank" rel="noreferrer">
          youtube.com
        </a>
        .
      </strong>
    </Alert>
  );
};

export const Popup: React.FC = () => {
  const { t } = useTranslation();
  const classes = useStyles();

  const version = config.VERSION;
  const timeago = formatDistance(
    parseISO(config.BUILD_DATE),
    new Date(),
    {
      addSuffix: true,
    }
  );

  return (
    <Card className={classes.container}>
      <Box
        className={classes.header}
        display="flex"
        justifyContent="center"
      >
        <a
          className={classes.link}
          href={config.WEB_URL}
          target="_blank"
          rel="noreferrer"
        >
          <img className={classes.img} src="/ycai-logo.svg" />
        </a>
      </Box>

      <WithQueries
        queries={{ settings }}
        render={QR.fold(
          () => (
            <PopupLoader />
          ),
          PopupErrorBox,
          ({ settings }): any => (
            <>
              <CardContent className={classes.content}>
                <Box
                  className={classes.version}
                  alignItems="center"
                  display="flex"
                  justifyContent="center"
                >
                  <Typography variant="caption">
                    {t('popup:version', { version, date: timeago })}
                  </Typography>
                  {config.NODE_ENV === 'development' ? (
                    <Typography
                      color="primary"
                      variant="subtitle1"
                      display="inline"
                      style={{
                        fontWeight: 800,
                        marginBottom: 0,
                      }}
                    >
                      &nbsp;DEVELOPMENT
                    </Typography>
                  ) : null}
                </Box>

                <Settings settings={settings} />
              </CardContent>
              <CardActions>
                <Button
                  className={classes.dashboardButton}
                  color="primary"
                  fullWidth
                  href={'/index.html'}
                  size="large"
                  target="_blank"
                  variant="contained"
                >
                  {t('dashboard:title')}
                </Button>
              </CardActions>
            </>
          )
        )}
      />
      {/* </Grid> */}
    </Card>
  );
};
