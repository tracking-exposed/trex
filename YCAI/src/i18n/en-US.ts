import { CustomTypeOptions } from 'react-i18next';

const resources: CustomTypeOptions['resources'] = {
  title: 'YouChoose AI',
  common: {
    title: 'YouChoose AI',
    description: 'YouChoose the recommendation model',
    coming_soon: 'Coming soon',
    empty_list: 'No {{resource}} found.',
    empty_string: '',
  },
  actions: {
    popup_bootstrap: 'Bootstrap',
    add: 'Add',
    copied: 'Copied!',
    drag_drop_recommendations:
      'Drag and drop to change the order of appearance',
    manage_recommendations: 'Manage recommendations',
    link_channel: 'Link channel',
    unlink_channel: 'Unlink Channel',
    unlink_channel_confirm_text:
      'Are you sure you want to unlink your channel from the YouChoose extension?\nYour recommendations will remain visible on the YouChoose platform.',
    unlink_channel_confirm_no: 'No, stay signed-in',
    unlink_channel_confirm_yes: 'Yes, unlink my channel',
    delete: 'Delete',
    clear: 'Clear',
    editThisVideo: 'Edit this video',
    importVideos: 'Import videos',
    addToCurrentVideo: 'Add to current video',
    compare: 'Compare',
    related: 'Related',
    next: 'Next',
    verify_channel: 'Verify',
    generate_keypair: 'Generate keypair',
    refresh_keypair: 'Refresh keypair',
    delete_keypair: 'Delete keypair',
    download_keypair: 'Download keypair',
    pull_creator_videos: 'Import your videos',
    update_creator_videos_list: 'Refresh',
    copy_verification_code: 'Copy code',
    unlink_profile: 'Unlink profile',
    edit_access_token: 'Edit Access Token',
    save_access_token: 'Save Access Token',
    download_access_token: 'Download Access Token',
    download: 'Download',
    move_recommendation_up: 'Move recommendation up',
    move_recommendation_down: 'Move recommendation down',
    delete_recommendation_button: 'Delete',
    edit_recommendation_form_title: 'Edit recommendation',
    edit_recommendation_button: 'Edit',
    edit_recommendation_description:
      'Explain briefly why you found this content relevant for your audience',
    cancel: 'Cancel',
    save: 'Save',
    manage_tokens: 'Manage tokens',
    reset_settings: 'Reset settings',
    reload_extension: 'Reload extension',
  },
  errors: {
    an_error_occurred: 'An error occurred',
  },
  routes: {
    lab_title: 'LAB - Choose the Recommendations on Your Videos',
    lab_title_short: 'LAB',
    lab_subtitle:
      'Click on a video to pick its recommendations.\nYour content, your choice!',
    lab_edit_subtitle:
      'Paste a link in the grey box below to add it to the recommendations of this video.\nYou can add recommendations towards any website!',
    recommendations_library_title_short: 'Recommendations',
    recommendations_library_subtitle: 'Recommendations Library',
    recommendations_library_title:
      'All the recommendations added by you will be linked to your videos',
    analytics: 'Analytics',
    settings: 'Settings',
    link_account:
      'Authenticate your Channel \n to start recommending on your videos',
  },
  account: {
    channel: 'Your channel URL or ID',
  },
  creator: {
    title: 'Creator',
  },
  hide_all: {
    title: 'Hide All',
  },
  link_account: {
    title: 'Authenticate your YouTube channel.',
    subtitle:
      'Follow these 2 steps in 1 min to start choosing recommendations on your videos:',
    label: 'Authenticate your YouTube channel',
    copy_verification_key:
      'Copy paste this unique code in your channel description',
    verification_code_hint:
      "<1>Click here to edit your channel description</1> from YouTube Studio.\n\nPaste the code anywhere in the description and click 'publish' on the top right.\n\nYou can remove this code after the verification is finished.",
    paste_channel_url: '- Paste the URL or ID of your YouTube Channel:',
    verification_failed: 'Oops, the channel authentication failed!',
    verification_failed_hint:
      'Please check that you have published the code\nin your channel description and try again.',
    go_back_to_step_one_hint:
      'If the verification keeps failing, <1>click CLEAR</1> and make sure\nyou have pasted the correct URL of your YouTube channel.',
    channel_not_found:
      "We couldn't find a channel with the ID you provided.\nPlease make sure the URL or ID you pasted is correct.",
    already_have_token:
      'Or <1>click here</1> if you already have an access token.',
    token_modal_title: 'Authenticate with an access token',
    token_modal_description:
      'If you already have an access token, paste it below:',
    token_modal_submit: 'Authenticate',
    token_authentication_failed:
      'Authentication failed, please check your access token.',
  },
  youtube: {
    title: 'Youtube',
  },
  recommendations: {
    added_to_video_title: 'Recommendations added',
    by_creator_title: "Author's recommendations",
    total: 'Total recommendations',
    url: 'Recommendation URL',
    yours: 'Your recommendations',
    no_items: 'Recommendations will appear here once added!',
    add_to_video: 'Add a recommendation to this video',
    url_placeholder: 'http://...',
    url_helper_text: 'Insert a link',
    title: 'Recommendation title',
    description: 'Recommendation description',
    missing_description: 'This recommendation does not have a description.',
  },
  dashboard: { title: 'Youchoose Creator Studio' },
  popup: { version: 'version {{version}} build {{date}}' },
  analytics: {
    subtitle: 'Statistics computed thanks to user data donations',
    recommendability_score_title: 'Recommendability Score',
    recommendability_score_subtitle:
      'How much does YouTube promote your content?',
    recommendations_title: 'Recommendations Analyzed',
    recommendations_for_other_channels: 'From other channels',
    total_recommendations: 'Total recommendations',
    total_metadata: 'Total metadata',
    notifications_title: 'Notifications',
    top_n_cc_related_to_your_channel:
      'Most recommended channels on your videos:',
    advertising_connected_to_your_videos_title: 'Advertising',
    advertising_connected_to_your_videos_subtitle: 'Ads ran on your videos',
    advertising_empty_data: 'No advertising data collected yet on your channel',
  },
  videos: {
    no_results: 'No videos found.',
    no_selected: 'No video selected',
    no_video_id: 'No video id found',
  },
  settings: {
    contentCreatorRecommendationLabel: 'Enhance Youtube experience',
    contentCreatorRecommendationHint:
      'Show YouChoose pop in the YouTube sidebar',
    communityRecommendationsLabel: 'Community',
    communityRecommendationsHint: 'Coming soon ðŸŒ»',
    contributeToIndependentStatsLabel: 'Donate Anonymous Data',
    contributeToIndependentStatsHint:
      'Share data about Youtube to help understand the AI',
    contributeToIndependentStatsShowUILabel: 'UI highlight for donation',
    contributeToIndependentStatsShowUIHint:
      'Highlight the UI of collected and donated elements',
    api_list_title: 'API List',
    encrypted_contributions_private_key:
      'You can download your private key in dashboard.',
    keypair_passphrase: 'Key pair passphrase',
    keypair_private_key: 'Key pair secret key',
    keypair_public_key: 'Key pair public key',
    keypair_title: 'Keypair for independent contribution',
    keypair_description:
      'These are the secure cryptographic keys that are used to encrypt the data that you share with YouChoose.\nOnly you have access to the secret key and without it nobody can link together the information you share.',
    access_token_title: 'Your access token',
    access_token_description:
      'This access token allows this browser to access your profile on YouChoose.\nYou can share it with team-members to give them access to the YouChoose dashboard.',
    access_token: 'Access Token',
    data_donation_learn_more: 'Learn more',
    nudge_donation_opt_in:
      'Please consider donating anonymous data to help us understand the YouTube algorithm.',
    nudge_learn_more: 'Learn more.',
    nudge_not_now: 'Not now',
    nudge_agree: 'I agree',
  },
  ytVideoPage: {
    firstTab: 'Creator Recommendations',
    secondTab: 'Community Recommendations',
    thirdTab: 'Youtube Recommendations',
    distractionFree: 'Enjoy a distraction-free experience',
    noCCRecommendations:
      'This creator has not uploaded customized recommendations yet. If you own this channel, you can do it <1>here</1>.',
  },
  swagger: {
    title: 'The YouChoose AI API workbench',
    description:
      'In this section dedicated to developers you can explore and try out all of the API routes used by YouChoose.<br>You can use these APIs outside of YouChoose as you please.',
  },
};

export default resources;
