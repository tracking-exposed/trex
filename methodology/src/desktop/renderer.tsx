import * as React from "react";
import * as ReactDOM from "react-dom";
import { ThemeProvider } from "@material-ui/core";
import { App } from "./App";
import { theme } from "./theme";
import debug from "debug";

const main = (): void => {
  debug.enable("@yttrex*");

  ReactDOM.render(
    <React.StrictMode>
      <ThemeProvider theme={theme}>
        <App />
      </ThemeProvider>
    </React.StrictMode>,
    document.getElementById("guardoni")
  );
};

main();
