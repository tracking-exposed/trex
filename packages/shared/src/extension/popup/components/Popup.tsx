import { Card } from '@material-ui/core';
import FormHelperText from '@material-ui/core/FormHelperText';
import { Alert, AlertTitle } from '@material-ui/lab';
import { localLookup, configUpdate } from '../../background/sendMessage';
import config from '../../config';
import log from '../../logger';
import DashboardLinks, { DashboardLink } from './DashboardLinks';
import InfoBox from './InfoBox';
import Settings, { SettingsProps } from './Settings';
import UserSettings from '../../models/UserSettings';
import * as React from 'react';
import moment from 'moment';

const styles = {
  width: '400px',
};

export interface PopupProps {
  platform: string;
  platformURL: string;
  logo: string;
  getLinks: (opts: { publicKey: string }) => DashboardLink[];
  settings: Pick<SettingsProps, 'enabled'>;
}

type PopupState =
  | { status: 'init' }
  | {
      status: 'loading';
    }
  | {
      status: 'error';
      error: Error;
    }
  | {
      status: 'done';
      payload: UserSettings;
    };

let localLookupInterval: any;
const Popup: React.FC<PopupProps> = ({
  platform,
  platformURL,
  settings,
  logo,
  getLinks,
}) => {
  const [userSettingsS, setUserSettingsState] = React.useState<PopupState>({
    status: 'init',
  });

  const handleLocalLookup = (): void => {
    localLookup(true, (response) => {
      if (response.type === 'Error') {
        setUserSettingsState({
          status: 'error',
          error: response.error,
        });
        log.error('could not get user settings %O', response.error);

        // localLookupInterval = setTimeout(() => {
        //   log.info('Refetching settings...');
        //   handleLocalLookup();
        // }, 2000);
        return;
      }

      if (localLookupInterval) {
        localLookupInterval.clear();
        localLookupInterval = undefined;
      }

      setUserSettingsState({ status: 'done', payload: response.result });
    });
  };

  const handleConfigChange = React.useCallback((s: UserSettings): void => {
    configUpdate(s, (r) => {
      if (r.type === 'Error') {
        setUserSettingsState({
          status: 'error',
          error: r.error,
        });
      } else {
        setUserSettingsState({
          status: 'done',
          payload: r.result,
        });
      }
    });
  }, []);

  React.useEffect(() => {
    if (userSettingsS.status === 'init') {
      handleLocalLookup();
    }
  }, []);

  const deltaMs = config.BUILD_DATE
    ? Date.now() - new Date(config.BUILD_DATE).getTime()
    : 0;

  const timeAgo = moment.duration(deltaMs).humanize();

  const content = React.useMemo(() => {
    if (userSettingsS.status === 'loading' || userSettingsS.status === 'init') {
      return (
        <Card style={styles}>
          <Alert severity="info">
            <AlertTitle>Loading</AlertTitle>
            Loading user settings
          </Alert>
        </Card>
      );
    }

    if (userSettingsS.status === 'error') {
      if (
        userSettingsS.error.message.includes(
          "Error during 'LocalLookup' on codec UserSettings validation"
        )
      ) {
        return (
          <Card style={styles}>
            <Alert severity={'info'}>
              Access to{' '}
              <a target="_blank" href={platformURL} rel="noreferrer">
                {platform}
              </a>
              .
            </Alert>
          </Card>
        );
      }
      return (
        <Card style={styles}>
          <Alert severity={'error'}>
            <AlertTitle>Error</AlertTitle>
            Something went wrong, sorry.
            <p>
              {userSettingsS.error.name}: {userSettingsS.error.message}
            </p>
          </Alert>
        </Card>
      );
    }

    return (
      <Card>
        <FormHelperText>{platform} TRex â€” main switch</FormHelperText>
        <Settings
          enabled={settings.enabled}
          settings={userSettingsS.payload}
          onSettingsChange={handleConfigChange}
        />
        <FormHelperText>Access to your data</FormHelperText>
        <DashboardLinks
          publicKey={userSettingsS.payload.publicKey}
          getLinks={getLinks}
        />
        <FormHelperText>About</FormHelperText>
        <InfoBox logo={logo} />
      </Card>
    );
  }, [userSettingsS]);

  return (
    <div style={styles}>
      {content}
      <small>
        version {config.VERSION}, released {timeAgo} ago
      </small>
    </div>
  );
};

export default Popup;
