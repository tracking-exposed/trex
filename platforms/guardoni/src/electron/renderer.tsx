import { CssBaseline, ThemeProvider } from '@material-ui/core';
import debug from 'debug';
import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { HashRouter } from 'react-router-dom';
import { App } from './app/App';
import { theme } from './app/theme';

const main = (): void => {
  debug.enable('@trex*');

  ReactDOM.render(
    <React.StrictMode>
      <HashRouter>
        <ThemeProvider theme={theme}>
          <CssBaseline />
          <App />
        </ThemeProvider>
      </HashRouter>
    </React.StrictMode>,
    document.getElementById('guardoni')
  );
};

main();
