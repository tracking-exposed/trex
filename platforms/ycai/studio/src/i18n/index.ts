import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';
import en from './en-US';


export default i18n.use(initReactI18next).init({
  resources: {
    en: en,
  },
  fallbackLng: 'en',
  interpolation: {
    escapeValue: false,
  },
});
