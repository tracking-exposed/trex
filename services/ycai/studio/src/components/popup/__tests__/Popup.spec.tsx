import React from 'react';

import '../../../i18n';
import { chrome } from 'jest-chrome';
import renderer from 'react-test-renderer';
import { Popup } from '../Popup';

const onMessageListener = jest.fn().mockImplementation((r, s, sendRes) => {
  sendRes({ active: true });
  return true;
});

describe('Popup', () => {
  test('Should mount the Popup', () => {
    chrome.runtime.onMessage.addListener(onMessageListener);
    const component = renderer.create(<Popup />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
