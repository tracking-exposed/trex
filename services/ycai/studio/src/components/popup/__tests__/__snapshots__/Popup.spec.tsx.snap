// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup Should mount the Popup 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-container-1 MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiBox-root MuiBox-root-9 makeStyles-header-3"
  >
    <a
      className="makeStyles-link-6"
      rel="noreferrer"
      target="_blank"
    >
      <img
        className="makeStyles-img-5"
        src="/ycai-logo.svg"
      />
    </a>
  </div>
  <div
    className="MuiPaper-root MuiAlert-root MuiAlert-standardInfo MuiPaper-elevation0"
    role="alert"
  >
    <div
      className="MuiAlert-icon"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20, 12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10, 10 0 0,0 12,2M11,17H13V11H11V17Z"
        />
      </svg>
    </div>
    <div
      className="MuiAlert-message"
    >
      <div
        className="MuiTypography-root MuiAlertTitle-root MuiTypography-body1 MuiTypography-gutterBottom"
      >
        Loading settings...
      </div>
      <strong>
        Access
         
        <a
          href="https://www.youtube.com"
          rel="noreferrer"
          target="_blank"
        >
          youtube.com
        </a>
        .
      </strong>
    </div>
  </div>
</div>
`;
